syft/file/cataloger/filemetadata/cataloger.go

```diff
@@ -20,52 +20,57 @@ func NewCataloger() *Cataloger {
 func (i *Cataloger) Catalog(resolver file.Resolver, coordinates ...file.Coordinates) (map[file.Coordinates]file.Metadata, error) {
 	results := make(map[file.Coordinates]file.Metadata)
 	var locations <-chan file.Location
-
 	if len(coordinates) == 0 {
 		locations = resolver.AllLocations()
 	} else {
 		locations = func() <-chan file.Location {
 			ch := make(chan file.Location)
 			go func() {
-				close(ch)
+				defer close(ch)
 				for _, c := range coordinates {
-					ch <- file.NewLocationFromCoordinates(c)
+					locs, err := resolver.FilesByPath(c.RealPath)
```

Comment:
this fixes the cataloger to always reference locations from the resolver, not to create new ones based on bespoke paths.

Reply:
I'll move this change to another PR

Reply:
https://github.com/anchore/syft/pull/2370